Sources/Player/OfflineEngine/Internal/MediaDownloader.swift

```diff
@@ -60,6 +60,43 @@ private extension MediaDownloader {
 	func createTask(for url: URL) -> URLSessionDownloadTask {
 		urlDownloadSession.downloadTask(with: url)
 	}
+
+	func calculateHLSStreamSize(for path: URL) -> UInt64 {
+		let fileManager = FileManager.default
+		var totalSize: UInt64 = 0
+
+		if let enumerator = fileManager.enumerator(
+			at: path,
+			includingPropertiesForKeys: [.fileSizeKey],
+			options: [],
+			errorHandler: nil
+		) {
+			for case let fileURL as URL in enumerator {
+				do {
+					let fileAttributes = try fileURL.resourceValues(forKeys: [.fileSizeKey])
+					if let fileSize = fileAttributes.fileSize {
+						totalSize += UInt64(fileSize)
+					}
+				} catch {
+					print("Error reading file size: \(error.localizedDescription)")
```

Comment:
[question] do we need proper log? I guess it depends on how this can this affect the offlining. (same for below)

Reply:
Addressed 👍